src/query/service/src/interpreters/interpreter_dictionary_show_create.rs

```diff
@@ -0,0 +1,207 @@
+        let tenant = self.ctx.get_tenant();
+        let catalog = self.ctx.get_catalog(self.plan.catalog.as_str()).await?;
+        let dict_name = self.plan.dictionary.clone();
+
+        let dict_ident = TenantDictionaryIdent::new(
+            tenant,
+            DictionaryIdentity::new(self.plan.database_id, dict_name.clone()),
+        );
+        let dictionary = if let Some(reply) = catalog.get_dictionary(dict_ident).await? {
+            reply.dictionary_meta
+        } else {
+            return Ok(PipelineBuildResult::create());
+        };
+        let settings = self.ctx.get_settings();
+        let settings = ShowCreateQuerySettings {
+            sql_dialect: settings.get_sql_dialect()?,
+            quoted_ident_case_sensitive: settings.get_quoted_ident_case_sensitive()?,
+        };
+
+        let create_query: String =
+            Self::show_create_query(catalog.as_ref(), &dictionary, &dict_name, &settings).await?;
+        let block = DataBlock::new(
+            vec![
+                BlockEntry::new(DataType::String, Value::Scalar(Scalar::String(dict_name))),
+                BlockEntry::new(
+                    DataType::String,
+                    Value::Scalar(Scalar::String(create_query)),
+                ),
+            ],
+            1,
+        );
+        PipelineBuildResult::from_blocks(vec![block])
+    }
+}
+
+impl ShowCreateDictionaryInterpreter {
+    pub async fn show_create_query(
+        _catalog: &dyn Catalog,
+        dictionary: &DictionaryMeta,
+        dict_name: &str,
+        settings: &ShowCreateQuerySettings,
+    ) -> Result<String> {
+        let sql_dialect = settings.sql_dialect;
+        let quoted_ident_case_sensitive = settings.quoted_ident_case_sensitive;
+        let schema = dictionary.schema.clone();
+        let n_fields = schema.fields().len();
+        let source = dictionary.source.clone();
+        let source_options = dictionary.options.clone();
+        let comment = dictionary.comment.clone();
+        let pk_id_list = dictionary.primary_column_ids.clone();
+        let field_comments = dictionary.field_comments.clone();
+
+        let mut dict_create_sql = format!(
+            "CREATE DICTIONARY {} (\n",
+            display_ident(dict_name, quoted_ident_case_sensitive, sql_dialect)
+        );
+
+        // Append columns and indexes.
+        {
+            let mut create_defs = vec![];
+            for (idx, field) in schema.fields().iter().enumerate() {
+                let nullable = if field.is_nullable() {
+                    " NULL".to_string()
+                } else {
+                    " NOT NULL".to_string()
+                };
+                let default_expr = match field.default_expr() {
+                    Some(expr) => {
+                        format!(" DEFAULT {expr}")
+                    }
+                    None => "".to_string(),
+                };
+                let computed_expr = match field.computed_expr() {
```

Comment:
`default_expr` and `computed_expr` can be removed, because dictionary don't allow have default value or computed expression.